v-next/core/src/internal/plugins/plugin-validation.ts

```diff
@@ -1,6 +1,110 @@
+import { createRequire } from "node:module";
+import path from "node:path";
+import process from "node:process";
+
+import { HardhatError } from "@nomicfoundation/hardhat-errors";
+import semver from "semver";
+
 import { HardhatPlugin } from "../../types/plugins.js";
 
-export async function validatePluginNpmDependencies(_plugin: HardhatPlugin) {
-  // TODO: If it has an npm package, validate their peer dependencies
-  // If any is missing, throw
+/**
+ * A simplified version of the package.json object
+ */
+interface PackageJson {
```

Comment:
you have this interface available in `hardhat-utils/package`

Reply:
I have pulled it in directly.